diff --git a/include/pangolin/platform.h b/include/pangolin/platform.h
index b5a223b7..c35bb20f 100644
--- a/include/pangolin/platform.h
+++ b/include/pangolin/platform.h
@@ -30,7 +30,9 @@
 #include <pangolin/config.h>
 
 // Include portable printf-style format macros
+#ifndef __STDC_FORMAT_MACROS
 #define __STDC_FORMAT_MACROS
+#endif
 
 #ifdef _GCC_
 #  define PANGOLIN_DEPRECATED __attribute__((deprecated))
diff --git a/include/pangolin/utils/picojson.h b/include/pangolin/utils/picojson.h
index 0a5cb9f7..cf47540e 100644
--- a/include/pangolin/utils/picojson.h
+++ b/include/pangolin/utils/picojson.h
@@ -69,7 +69,9 @@ extern "C" {
 
 // experimental support for int64_t (see README.mkdn for detail)
 #ifdef PICOJSON_USE_INT64
+#ifndef __STDC_FORMAT_MACROS
 # define __STDC_FORMAT_MACROS
+#endif
 # include <errno.h>
 # include <inttypes.h>
 #endif // PICOJSON_USE_INT64
